pepper-apis/src/main/java/org/broadinstitute/ddp/route/PatchFormAnswersRoute.java

```diff
@@ -175,6 +175,16 @@ public Object handle(Request request, Response response) {
                     Question question = handle.attach(QuestionDao.class).getQuestionByActivityInstanceAndDto(questionDto,
                             instanceGuid, false, languageCodeId);
 
+                    //validation to check if question is a composite child
+                    Optional<Long> parentQuestionId = handle.attach(JdbiCompositeQuestion.class)
+                            .findParentQuestionIdByChildQuestionId(question.getQuestionId());
+                    if (parentQuestionId.isPresent()) {
+                        LOG.warn("Passed question stable ID : " + questionStableId + " is a Composite child question. "
+                                + "Only entire Composite question answer can be updated ");
+                        throw ResponseUtil.haltError(response, HttpStatus.SC_NOT_FOUND, new ApiError(
+                                ErrorCodes.QUESTION_REQUIREMENTS_NOT_MET, "Only entire Composite question answer can be updated"));
```

Comment:
ErrorCodes.QUESTION_REQUIREMENTS_NOT_MET or any other constant like: OPERATION_NOT_PERMITTED or something else ?
Also return code: 404 or something else !!

Reply:
I think `OPERATION_NOT_PERMITTED` makes sense here. How about we go with `400`?

Reply:
I agree with `OPERATION_NOT_PERMITTED`, but I think a `404` would be better suited. There isn't a client error with the request, the child question just doesn't exit from that perspective.

Reply:
If we returned `404`, I will probably expect to see something like `... hey, we can't find that question!` But instead here we're saying that whatever the client is trying to do is not permitted and that a composite question answer needs to be updated as a whole, so I'll lean more towards `400`.

Reply:
I can average it and make it 402 😁

Reply:
In the ticket Implementation sketch, Marco suggested 404 😺

Reply:
If folks are vouching for `404`, then let's use `404`. In that case, may I suggest returning `QUESTION_NOT_FOUND` instead of `OPERATION_NOT_PERMITTED`?

Reply:
@bskinner Are you ok with return error code: QUESTION_NOT_FOUND ?

Reply:
That works great with me!

Reply:
One final look please